src/PageHeader/manageResponsiveValueWithCSS.ts

```diff
@@ -0,0 +1,130 @@
+import {isResponsiveValue, ResponsiveValue} from '../hooks/useResponsiveValue'
+import {BetterSystemStyleObject} from '../sx'
+
+function areAllValuesTheSame(responsiveValue: ResponsiveValue<boolean | number | string>): boolean {
+  if ('narrow' in responsiveValue && 'regular' in responsiveValue && 'wide' in responsiveValue) {
+    const values = Object.values(responsiveValue)
+    return values.every(value => value === values[0])
+  }
+  return false
+}
+function haveRegularAndWideSameValue(responsiveValue: ResponsiveValue<boolean | number | string>): boolean {
+  if ('regular' in responsiveValue && 'wide' in responsiveValue) {
+    return responsiveValue.regular === responsiveValue.wide
+  }
+  return false
+}
+
+/**
+ * This function is inspired by the `useResponsiveValue` hook and it's used to render responsive values with CSS.
+ * @param value - The value that needs to be rendered responsively
+ * @param cssProperty - The CSS property whoes value needs to be rendered responsively
+ * @param comparisonFunc - A function that evaluates which value to assign to the CSS property
+ 
+ * @example
+ * CSSManagedResponsiveValue({narrow: true, regular: true, wide: false}, 'display', value => {
+      return value ? 'none' : 'flex'
+    })
+ * @returns
+ * {
+ *   "@media screen and (max-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 768px)": {
+ *     "display": "none"
+ *   },
+ *   "@media screen and (min-width: 1440px)": {
+ *     "display": "flex"
+ *   }
+ * }
+ * 
+ * * @example
+ * CSSManagedResponsiveValue({regular: 'border.default', wide: 'canvas.inset'}, 'backgroundColor', (value): string => {
+    return value
+  })
+ * @returns
+ * {
+ *   "@media screen and (min-width: 768px)": {
+ *     "backgroundColor": "border.default"
+ *   },
+ *   "@media screen and (min-width: 1440px)": {
+ *     "backgroundColor": "canvas.inset"
+ *   }
+ * }
+ * 
+ * * @example
+* CSSManagedResponsiveValue({narrow: 'filled', regular: 'line'}, 'height', (value): string => {
+    return {
+      filled: 8,
+      line: 1,
+    }[value]
+  })
+ * @returns
+ * {
+ *  "@media screen and (max-width: 768px)": {
+ *     "height": 8
+ *   }
+ *   "@media screen and (min-width: 768px)": {
+ *     "height": 1
+ *   },
+ * }
+ */
+export function CSSManagedResponsiveValue<T>(
```

Comment:
I'm not sure about the name `CSSManagedResponsiveValue`. It seems a little long and unclear. The word "value" feels misleading because it actually returns an object with the property and the value (not just the value). How do you feel about calling it something like `responsiveCssProperty`?

Reply:
Yes, I agree the name is unclear and too long! Josh also suggested `getBreakpointValues` and your comment about `value` makes so much sense as well. I love both of your suggestions! Maybe using `CSSProperty` is a bit misleading as well because it doesn't mention about the value?  Let me try to combine both of your feedback. 

`CSSProperty + Values = CSS Declarations`

I feel like `responsive` and `breakpoints` could be interchangeable. 
if we want to be more specific about what we return maybe we could use `getBreakpointDeclarations`?
If we want to be more consistent with the hook we could use `responsiveCSSDeclarations`?

If it hard! 😬 Let me know what you think!

Reply:
`getBreakpointDeclarations` sounds good to me 👍